src/App.js

```diff
@@ -2,89 +2,171 @@ import React, { useState } from "react";
-      alert("Please enter details !!!");
+      if (event.key === "Enter") {
+        setTodos([
+          {
+            id: new Date().getTime(),
+            TodoInputValue: inputValue,
+            isCompleted: false,
+          },
+          ...todos,
+        ]);
+        setInputValue("");
+      }
     }
-    // }
   };
 
   // ***************  Handle Delete Todo  ***********************
 
   const HandleDelete = (id) => {
-    const filtertodo = todos.filter((todo) => todo.id !== id);
-    settodos(filtertodo);
+    setTodos(todos.filter((todo) => todo.id !== id));
   };
 
   // ***************  Handle Check Todo  ***********************
 
   const HandleCheck = (id) => {
-    const updatedtodos = todos.map((todo) => {
+    const originalTodo = todos.slice();
+    const updatedTodos = originalTodo.map((todo) => {
       if (todo.id === id) {
-        todo.status = !todo.status;
+        todo.isCompleted = !todo.isCompleted;
       }
       return todo;
     });
-    settodos(updatedtodos);
+    
+    // *************  Checking if Single todo is checked or not  ****************
+    const SearchSingleCheck = updatedTodos.find(todo => todo.isCompleted === true);
+    if(SearchSingleCheck){
+      setSingleCheck(true);
+    }
+    else{
+      setSingleCheck(false);
+    }
+
+    // **********  Sortings as per isCompleted  ***********
+    setTodos(updatedTodos.sort((x) => (x.isCompleted ? 1 : -1)));
   };
 
-  // ***************  Handle Check Todo  ***********************
+  // ***************  Handle Edit Todo  ***********************
 
-  const HandleEdit = (id, newinput) => {
-    const updatedtodos = todos.map((todo) => {
+  const HandleEdit = (id, newInput) => {
+    setSingleEdit(true);
+    const originalTodo = todos.slice();
+    const updatedTodos = originalTodo.map((todo) => {
       if (todo.id === id) {
-        todo.value = newinput;
+        todo.TodoInputValue = newInput;
       }
       return todo;
     });
-    settodos(updatedtodos);
+    setTodos(updatedTodos);
   };
 
+  // ***************  Handle All Clear Todo  ********************
+
+  const HandleAllClear = () => {
+    setTodos([]);
+    setInputValue("");
+    setSelectedCheck(false);
+    setError(false);
+    setSingleCheck(false);
+  };
+
+
+  // ***************  Handle Selected Checkbox  *******************
+
+  const HandleSelectedCheckbox = () =>{
+    setSelectedCheck(!selectedCheck);
+    const originalTodo = todos.slice();
+    const updatedTodo = originalTodo.map(todo => {
+       if(selectedCheck === false){
+         todo.isCompleted = true;
+       }
+       else{
+         todo.isCompleted = false;
+         setSingleCheck(false);
+       }
+       return todo
+      });
+
+     setTodos(updatedTodo);
+  }
+
+  // **************** Handle Selected Clear  **********************
+
+  const HandleSelectedClear = () =>{
+    const originalTodo = todos.slice();
+    const updatedTodo = originalTodo.filter(todo => todo.isCompleted !== true);
```

Comment:
Here we don't need `originalTodo` as filter will clear new array & we aren't changing any value in callback